.github/workflows/instructlab-knowledge-e2e.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow is designed to test the notebooks in "notebooks/instructlab-knowledge"
+# This job can be coppied for end to end tests of notebooks in other working directories
+
+name: Instructlab Knowledge End to End Tests
+
+# trigger on open pull requests only
+on:
+    pull_request:
+        branches:
+          - "main"
+        types: [opened, reopened]
+
+env:
+    # dependencies that need to be installed using pip for testing to work
+    TESTING_PIP_DEPENDENCIES: "papermill nbformat"
+
+jobs:
+    end_to_end:
+        strategy:
+            matrix:
+                # add notebooks within the "notebooks/instructlab-knowledge" repo to end to end test here
+                notebooks_to_test: ["instructlab-knowledge.ipynb"]
+        runs-on: ubuntu-latest
+        env:
+            # customize the workflow here
+            API_KEY: ""
+            API_URL: "http://127.0.0.1:11434/v1"
+            MODEL_ID: "Mixtral-8x7B" # must be open-AI compatible when using inference mock
+        steps:
+            - uses: actions/checkout@v3
+            - name: Set up Python
+              uses: actions/setup-python@v3s
+              with:
+                python-version: '3.11'
+                cache: pip
+            - name: Install Testing Tools
+              run: pip install ${{ env.TESTING_PIP_DEPENDENCIES }}
+            - name: Validate The Notebook
+              working-directory: ./notebooks/instructlab-knowledge
+              run: |
+                git clone https://github.com/instructlab/instructlab.git
+                python3 instructlab/scripts/validate_notebook.py ${{ matrix.notebooks_to_test }}
```

Comment:
Do we expect this to be used anywhere but CI? If not, let's pick a dedicated directory that is specifically and only for CI-related tooling.

Reply:
WIP already: https://github.com/instructlab/ci-actions/pull/15